src/compiler/parser.ts

```diff
@@ -4720,10 +4720,16 @@ module ts {
                         amdModuleName = amdModuleNameMatchResult[2];
                     }
 
-                    var amdDependencyRegEx = /^\/\/\/\s*<amd-dependency\s+path\s*=\s*('|")(.+?)\1/gim;
+                    var amdDependencyRegEx = /^\/\/\/\s*<amd-dependency/gim;
+                    var pathRegex = /path\s*=\s*('|")(.+?)\1/gim
+                    var nameRegex = /name\s*=\s*('|")(.+?)\1/gim
                     var amdDependencyMatchResult = amdDependencyRegEx.exec(comment);
                     if (amdDependencyMatchResult) {
-                        amdDependencies.push(amdDependencyMatchResult[2]);
+                        var pathMatchResult = pathRegex.exec(comment)
+                        var nameMatchResult = nameRegex.exec(comment)
+                        if(pathMatchResult){
+                            amdDependencies.push({path:pathMatchResult[2], name:nameMatchResult?nameMatchResult[2]:null});

```

Comment:
Space after `{`, before `}`, before/after `?` and `:`, use `undefined` instead of `null`.